java/memory/src/main/java/org/apache/arrow/memory/BaseAllocator.java

```diff
@@ -111,15 +112,15 @@ protected BaseAllocator(
       historicalLog = null;
       childLedgers = null;
     }
-    this.roundingPolicy = roundingPolicy;
+    this.roundingPolicy = config.getRoundingPolicy();
   }
 
   AllocationListener getListener() {
     return listener;
   }
 
   @Override
-  public BufferAllocator getParentAllocator() {
+  public BaseAllocator getParentAllocator() {
```

Comment:
It is okay to expose here as long as the BufferAllocator interface is BufferAllocator.